lib/asciidoctor/pdf/converter.rb

```diff
@@ -2180,10 +2180,23 @@ def convert_table node
           head_border_bottom_width = theme.table_head_border_bottom_width || table_border_width
         end
         [:top, :bottom, :left, :right].each {|edge| border_width[edge] = table_border_width }
+
         table_grid_color = theme.table_grid_color || table_border_color
-        table_grid_style = (theme.table_grid_style || table_border_style).to_sym
+        table_grid_style = theme.table_grid_style || table_border_style
+        if table_grid_style.is_a? Array
+          raise 'table-grid-style array must contain two styles' if table_grid_style.size != 2
+          table_grid_style = [table_grid_style[0].to_sym, table_grid_style[1].to_sym]
+        else
+          table_grid_style = Array.new 2, table_grid_style.to_sym
+        end
         table_grid_width = theme.table_grid_width || theme.table_border_width
-        [:cols, :rows].each {|edge| border_width[edge] = table_grid_width }
+        if table_grid_width.is_a? Array
+          raise 'table-grid-width array must contain two numbers separated by comma, e.g [0.5,1]' if table_grid_width.size != 2 || table_grid_width.detect {|x| !x.is_a? Numeric }
```

Comment:
Same, we don't need the validation here. I prefer to do these sorts of checks in the theme, if we even do them. I defer a lot of them because we expect the theme to be correct. A crash is the right behavior if it is not.

Reply:
[kjell@p10s asciidoctor-pdf]$ bin/asciidoctor-pdf -a pdf-theme=theme.yml test.adoc
comparison of String with 0 failed
  Use --trace for backtrace

This is the error I get if I now try to use something not a number in the style.  But I guess you are right that those who make themes can be thought of as experts.  My check only handled the [] case anyway, so it is now removed.

Reply:
> I guess you are right that those who make themes can be thought of as experts.

I'm not going to go quite that far. I just don't think the point of usage is the right place for validations because it muddles up the code and is not likely to be comprehensive anyway. It would either be in the theme loader or some sort of theme validator.

Reply:
So a separate issue for adding sanity checks to the theme_loader.rb then?

Reply:
Yes. Though it will need to be driven by someone in the community because I'm not able to work on something like that right now.